### PR TITLE
Rework search team goals

### PR DESCRIPTION
**Pull request by lguychard**
- Less goals
- Goals to no longer nest under "vision" headers
- Goals have targets (ideally metrics-based), with an attached timeline, and progress updates (a bunch of TODOs here)
- Goals problem sections list to customer examples (a bunch of TODOs here also)

### CODE REVIEW

**Comment by poojaj-tech** [comment]
Thanks for creating this. I like this break down for the goals and the goal coverage for the WIP work. 

My initial feedback is more around naming and structure.  We can continue iterating on the content in future PRs.

**Comment by poojaj-tech** [comment]
(I'd forgotten to click submit after I entered in my review comments, my comment from 15 hrs ago refers to the feedback just submitted 🤦‍♂️)

File: handbook/engineering/search/goals.md [hunk 1]
```diff
@@ -4,48 +4,48 @@
...
-      - It is hard for users to quickly get to code they care about.
-          - **In progress work:** [Enterprise homepage](#1-enterprise-homepage)
-  - **Outcomes:**
-     - New users introduced to Sourcegraph are able to quickly run searches that show them the value of Sourcegraph.
-     - Users can run searches over code they care about more quickly.
-     - Sourcegraph surfaces code and searches users care about.
+**Problem:** Search is too slow or frequently times out at the scale some of our customers care about, and at the scale we want Sourcegraph Cloud to reach. In large-scale codebases, running the same search several times yields varying result counts, with no insights into the underlying reasons. Customers cannot trust that their codebase was exhaustively searched.
 
-### Shareable
+**Targets**
```
[the remark this answers is not in the document]
Comment (lguychard) [reply handbook/engineering/search/goals.md]: Reworded to "success criteria" here: https://github.com/sourcegraph/about/pull/1954/commits/9a748864d36360c386632c8780881bfdd16bc403

File: handbook/engineering/search/goals.md [hunk 2]
```diff
@@ -4,48 +4,48 @@
...
-  - **Problem:** Users want to be notified about important things going on in their code.
-  - **Outcome:** Notifications create a shared understanding and raise awareness of what’s going on in the code.
-  - **Planned work:** [Code monitoring version 1](#6-code-monitoring-version-1)
+- **By EOY 2020**: Latency for the set of search queries run by search-blitz is less than TODO Nms with 500k repositories indexed on Sourcegraph Cloud
+    - 2020-11-05: TODO
+- **By EOY 2020**: P99 time to first result is less than TODO Nms across all search types on Sourcegraph Cloud
+    - 2020-11-05: TODO
+
+
+**Milestones**
```
[the remark this answers is not in the document]
Comment (lguychard) [reply handbook/engineering/search/goals.md]: Reworded to "planned work": https://github.com/sourcegraph/about/pull/1954/commits/1ae9917657882c517f2e663700ef993d6513e48f

File: handbook/engineering/search/goals.md [hunk 3]
```diff
@@ -4,48 +4,48 @@
...
 Goals help us achieve our [vision](index.md#vision) for search.
 
-### Scalable
+### Mature query language
 
-- **Scale indexed search to 500k repositories**
-   - **Problem:** We have customers who need Sourcegraph to scale to 500k repositories. Some parts of Sourcegraph don't work well at that scale.
-   - **Outcome:** Sourcegraph can search 500k repositories in less than 300ms. This is on par with [grep.app](https://grep.app).
-   - **Planned work:** [Scaling indexed text search to 500k repositories](#4-scaling-indexed-text-search-to-500k-repositories)
+**Problem:** customers choose other search tools over Sourcegraph because they cannot express the queries they care about with Sourcegraph. TODO customer examples.
```
Comment (christinaforney) [inline handbook/engineering/search/goals.md:9]: The problem was never that customers chose other search tools. It was more about being able to be a competitive solution for customers using other search solutions, who weren't happy with those solutions either. Either way, I believe we've gotten past this point with the bulk of AND/OR query work. In addition to this prior motivation, we had many requests to provide custom search filters to enable different complex queries, which we previously solved by creating the new filters for each. Adding more filters is not a long term solution, and there needs to be a way to construct these complex queries.

```suggestion
**Problem:** The Sourcegraph query language does not support the growing complexity of searches being run by our users.

Examples:

- I want to only show examples from repositories that have been modified in the last 3 months so I see the most recent solutions.
- I want to find repositories that have not been modified in over a year so I can deprecate them.
- TODO: add more examples.
```

File: handbook/engineering/search/goals.md [hunk 4]
```diff
@@ -4,48 +4,48 @@
...
 
-- **Extend Search Query Language**
-  - **Problem:** It is imperative that users can find and filter the code, files, repositories, and commits they care about in extremely large codebases.
-  - **Outcomes:** Users can express relations on code, files, repositories, and commits in search queries to more effectively filter the data they need. Sourcegraph is the exclusive industrial-strength search solution that provides these capabilities.
-  - **Planned work:** [Search language rules engine](#10-search-language-rules-engine)
+1. Search expressions
+1. Search language rules engine
 
-### Easy to use
+### Performant search users can trust
```
[the remark this answers is not in the document]
Comment (christinaforney) [reply handbook/engineering/search/goals.md:20]:
```suggestion
### Users trust Sourcegraph for fast and reliable search results at scale
```

File: handbook/engineering/search/goals.md [hunk 5]
```diff
@@ -4,48 +4,48 @@
...
+- **By EOY 2020**: P99 time to first result is less than TODO Nms across all search types on Sourcegraph Cloud
+    - 2020-11-05: TODO
+
+
+**Planned work**
+
+1. Scale indexed text search to 500k repositories
+1. Streaming search
+
+### Sourcegraph helps customers monitor important changes to their codeebase
```
Comment (christinaforney) [inline handbook/engineering/search/goals.md:37]: Could build on this to make it a bit more grand and a larger vision:

```suggestion
### Sourcegraph is part of the development workflow and proactively notifies of important code changes
```

**Review by christinaforney — APPROVED** [review]
Made some suggestions to headings and problem statements. Feel free to ignore if you disagree, and please merge so we can iterate :)

File: handbook/engineering/search/goals.md (hunk 3 above)
Comment (lguychard) [reply handbook/engineering/search/goals.md:9]: Great points, thanks!

File: handbook/engineering/search/goals.md (hunk 5 above)
Comment (lguychard) [reply handbook/engineering/search/goals.md:43]: I do like the "proactive" part, I'm less sold on the "part of the development workflow". Cutting the apple in half and at least fixing the typo :)

```suggestion
### Sourcegraph proactively notifies of important code changes
```

**Pull request merged**